ompi/mca/coll/basic/coll_basic_reduce_scatter_block.c

```diff
@@ -105,8 +106,11 @@ mca_coll_basic_reduce_scatter_block_inter(const void *sbuf, void *rbuf, int rcou
         span = opal_datatype_span(&dtype->super, totalcounts, &gap);
 
         tmpbuf = (char *) malloc(span);
+        if (NULL == tmpbuf) {
+            return OMPI_ERR_OUT_OF_RESOURCE;
+        }
         tmpbuf2 = (char *) malloc(span);
-        if (NULL == tmpbuf || NULL == tmpbuf2) {
+        if (NULL == tmpbuf2) {
             return OMPI_ERR_OUT_OF_RESOURCE;
```

Comment:
We are leaking tmp here. I would rewrite the code to handle both buffers into a single allocation.
```c
tmpbuf = (char*)malloc(2*span);
if(NULL == tmpbuf) {
    return OMPI_ERR_OUT_OF_RESOURCE;
}
tmpbuf2 = tmpbuf + span;
```
Maybe with a realignment of tmp2 to 8 bytes.

Reply:
Doh. Can't believe I forgot the free as that was the whole point. I changed it to a single allocation.